src/lib/components/TokenSelect/useQueriedTokenList.ts

```diff
@@ -0,0 +1,33 @@
+import useDebounce from 'hooks/useDebounce'
+import useActiveWeb3React from 'lib/hooks/useActiveWeb3React'
+import useTokenList from 'lib/hooks/useTokenList'
+import { useMemo } from 'react'
+import { useTokenBalances } from 'state/wallet/hooks'
+
+import { createTokenFilterFunction, tokenComparator, useSortedTokensByQuery } from './utils'
+
+// TODO: Include balance
+
+export default function useQueriedTokenList(query: string) {
+  const tokenMap = useTokenList()
+  const tokens = useMemo(() => Object.values(tokenMap).map(({ token }) => token), [tokenMap])
+
+  // Sorts tokens
+  const { account } = useActiveWeb3React()
+  const balances = useTokenBalances(account, tokens)
+  const comparator = useMemo(() => tokenComparator.bind(null, balances), [balances])
+  const sortedTokens = tokens.sort(comparator)
+
+  // Filters tokens
+  const debouncedQuery = useDebounce(query, 200)
+  const filteredTokens = useMemo(
+    () => sortedTokens.filter(createTokenFilterFunction(debouncedQuery)),
+    [debouncedQuery, sortedTokens]
+  )
+
+  // Re-sorts tokens (ie to bump up exact matches)
+  const sortedFilteredTokens = useSortedTokensByQuery(filteredTokens, debouncedQuery)
+
+  // TODO(zzmp): Include native Currency
```

Comment:
This is a requirement for merge right? If app will be using this

Reply:
This is a requirement for launch, but may be deferred to a later PR